core/templates/components/forms/custom-forms-directives/thumbnail-display.component.spec.ts

```diff
@@ -0,0 +1,101 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @fileoverview Unit tests for Thumbnail Display component.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { async, ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+import { DomSanitizer } from '@angular/platform-browser';
+
+import { ThumbnailDisplayComponent } from './thumbnail-display.component';
+
+describe('Thumbnail Component', () => {
+  let component: ThumbnailDisplayComponent;
+  let fixture: ComponentFixture<ThumbnailDisplayComponent>;
+  class MockDomSanitizer {
+    bypassSecurityTrustResourceUrl(str: string): string {
+      return str;
+    }
+  }
+  /**
+   * Safe SVG Decoded
+   * <svg version="1.1" baseProfile="full" xmlns="http://www.w3.org/2000/svg">
+   *  <polygon id="triangle"
+   *           points="0,0 0,50 50,0" fill="#009900" stroke="#004400" />
+   * </svg>
+   */
+
+  const safeSvg = (
+    'data:image/svg+xml;base64,PHN2ZyBpZD0ic291cmNlIiB2ZXJzaW9uPSIxLjEiIGJhc2' +
+    'VQcm9maWxlPSJmdWxsIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgogID' +
+    'xwb2x5Z29uIGlkPSJ0cmlhbmdsZSIgcG9pbnRzPSIwLDAgMCw1MCA1MCwwIiBmaWxsPSIjMD' +
+    'A5OTAwIiBzdHJva2U9IiMwMDQ0MDAiPjwvcG9seWdvbj4KPC9zdmc+'
+  );
+
+  /**
+   * Malicious SVG Decoded
+   * <svg version="1.1" baseProfile="full" xmlns="http://www.w3.org/2000/svg">
+   * <polygon id="triangle"
+   *           points="0,0 0,50 50,0" fill="#009900" stroke="#004400" />
+   * <script type="text/javascript">
+   *   alert('This app is probably vulnerable to XSS attacks!');
+   * </script>
+   * </svg>
+   */
+  const maliciousSvg = (
+    'data:image/svg+xml;base64,PHN2ZyBpZD0ic291cmNlIiB2ZXJzaW9uPSIxLjEiIGJhc2' +
+    'VQcm9maWxlPSJmdWxsIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPgogID' +
+    'xwb2x5Z29uIGlkPSJ0cmlhbmdsZSIgcG9pbnRzPSIwLDAgMCw1MCA1MCwwIiBmaWxsPSIjMD' +
+    'A5OTAwIiBzdHJva2U9IiMwMDQ0MDAiPjwvcG9seWdvbj4KICA8c2NyaXB0IHR5cGU9InRleH' +
+    'QvamF2YXNjcmlwdCI+CiAgICBhbGVydCgnVGhpcyBhcHAgaXMgcHJvYmFibHkgdnVsbmVyYW' +
+    'JsZSB0byBYU1MgYXR0YWNrcyEnKTsKICA8L3NjcmlwdD4KPC9zdmc+'
+  );
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      declarations: [
+        ThumbnailDisplayComponent
+      ],
+      providers: [
+        {
+          provide: DomSanitizer,
+          useClass: MockDomSanitizer
+        }
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ThumbnailDisplayComponent);
+    component = fixture.componentInstance;
+  }));
+
+  it('should not render malicious SVG\'s on Init', fakeAsync(() => {
```

Comment:
@seanlip @kevintab95 @nithusha21 can you think of any other tests/ cases we could add here?

Reply:
Hm, what about the case where imgSrc is not a valid base64 encoded image.

Reply:
Done!

Reply:
Where? I don't see it in this file.

Reply:
It is in the spec file for svg sanitizer. https://github.com/oppia/oppia/pull/11306/files#diff-5ebc5b08264d68ccf490a885923458f7b9fdc13236a7fda9900ff3beb51d50f8R173-R174

Reply:
Ah ok, thanks!